modules/message-lane/src/benchmarking.rs

```diff
@@ -28,13 +28,25 @@ use frame_system::RawOrigin;
 use sp_std::{collections::btree_map::BTreeMap, convert::TryInto, ops::RangeInclusive, prelude::*};
 
 /// Fee paid by submitter for single message delivery.
-const MESSAGE_FEE: u32 = 1_000_000;
+pub const MESSAGE_FEE: u64 = 10_000_000_000;
 
 const SEED: u32 = 0;
 
 /// Module we're benchmarking here.
 pub struct Module<T: Config<I>, I: crate::Instance>(crate::Module<T, I>);
 
+/// Proof size requirements.
+pub enum ProofSize {
```

Comment:
I don't quite understand the difference between `HasExtraNodes` and `HasLargeLeaf` - can you give me an example or something to help clarify?

Reply:
Imagine binary tree and operation `binary_tree.encode()` that avoids any optimizations and encodes nodes, not values - i.e. pointers are also included in in the encoding. So there are two kinds of grow: (1) simply add 1 large value to the tree => since we include values into encoding, encoded value will be larger (2) insert many 0-byte values => number of nodes will grow and since we put internal information in the encoding, encoded value will be larger. So `HasExtraNodes` is the latter and `HasLargeLeaf` is the former. With these benchmarks, you may see that extra-nodes way has larger impact on performance, but the difference is not that big.